static/src/javascripts/projects/commercial/modules/header-bidding/prebid/prebid.spec.js

```diff
@@ -164,109 +169,138 @@ describe('initialise', () => {
+        const rtcData = window.pbjs.getConfig('realTimeData').dataProviders[0];
+		expect(rtcData.name).toEqual('permutive');
+		expect(rtcData.params.acBidders).toEqual([
+			'appnexus',
+			'ozone',
+			'pubmatic',
+			'trustx',
+		]);
+	});
+
+    test.each([
+        [true, false],
+        [false, true],
+        [false, false],
+    ])('should not generate RTD when Permutive is %s and prebidPermutiveAudience is %s', (p, a) => {
+        config.set('switches.permutive', p);
+		config.set('switches.prebidPermutiveAudience', a);
+		prebid.initialise(window);
+        const rtcData = window.pbjs.getConfig('realTimeData');
+		expect(rtcData).toBeUndefined();
+	});
+
+    describe('Prebid.js bidWon Events', () => {
+        test('should respond for correct configuration', () => {
+            let bidWonEventName;
+            let bidWonEventHandler;
+            const dummyAdvert = {
+                size: [200, 200],
+                hasPrebidSize: false,
+            };
+
+            window.pbjs.onEvent = jest.fn((eventName, eventHandler) => {
+                bidWonEventName = eventName;
+                bidWonEventHandler = eventHandler;
+            });
 
-        expect(bidWonEventName).toBe('bidWon');
-        expect(window.pbjs.onEvent).toHaveBeenCalledTimes(1);
+            getAdvertById.mockImplementation(() => dummyAdvert);
 
-        if (bidWonEventHandler) {
-            bidWonEventHandler({
-                height: 100,
-                width: 100,
-                adUnitCode: 'foo',
-            });
-        }
+            prebid.initialise(window);
 
-        expect(getAdvertById).toHaveBeenCalledTimes(1);
-        expect(getAdvertById).toHaveBeenCalledWith('foo');
-        expect(dummyAdvert.size).toMatchObject([100, 100]);
-        expect(dummyAdvert.hasPrebidSize).toBe(true);
-    });
+            expect(bidWonEventName).toBe('bidWon');
+            expect(window.pbjs.onEvent).toHaveBeenCalledTimes(1);
 
-    test('should not respond to prebid.js bidWon event if height missing from prebid data', () => {
-        let bidWonEventName;
-        let bidWonEventHandler;
+            if (bidWonEventHandler) {
+                bidWonEventHandler({
+                    height: 100,
+                    width: 100,
+                    adUnitCode: 'foo',
+                });
+            }
 
-        window.pbjs.onEvent = jest.fn((eventName, eventHandler) => {
-            bidWonEventName = eventName;
-            bidWonEventHandler = eventHandler;
+            expect(getAdvertById).toHaveBeenCalledTimes(1);
+            expect(getAdvertById).toHaveBeenCalledWith('foo');
+            expect(dummyAdvert.size).toMatchObject([100, 100]);
+            expect(dummyAdvert.hasPrebidSize).toBe(true);
         });
 
-        prebid.initialise(window);
-
-        expect(bidWonEventName).toBe('bidWon');
-        expect(window.pbjs.onEvent).toHaveBeenCalledTimes(1);
+        test('should not respond if height missing from prebid data', () => {
```

Comment:
Could you potentially macro the three 'should not respond' tests: height, width, adUnitCode?
On a quick a look it looks like the only difference is the object returned from bidWonEventHandler ?

Reply:
Good shout.